plugins/plugin-factory-github/codenvy-plugin-factory-github-server/src/main/java/com/codenvy/plugin/factory/github/server/GithubFactoryParametersResolver.java

```diff
@@ -0,0 +1,194 @@
+     *         map of parameters dedicated to factories
+     * @return true if it will be accepted by the resolver implementation or false if it is not accepted
+     */
+    @Override
+    public boolean accept(@NotNull Map<String, String> factoryParameters) {
+        // Check if url parameter is a github URL
+        return factoryParameters.containsKey(URL_PARAMETER_NAME) && GITHUB_PATTERN.matcher(factoryParameters.get(URL_PARAMETER_NAME)).matches();
+    }
+
+    /**
+     * Create factory object based on provided parameters
+     *
+     * @param factoryParameters
+     *         map containing factory data parameters provided through URL
+     * @throws BadRequestException when data are invalid
+     */
+    @Override
+    public Factory createFactory(@NotNull Map<String, String> factoryParameters) throws BadRequestException {
+
+        String githubUrl = factoryParameters.get("url");
+
+        // Apply github url to the regexp
+        Matcher matcher = GITHUB_PATTERN.matcher(githubUrl);
+        if (!matcher.matches()) {
+            throw new BadRequestException(String.format(
+                    "The given github url %s is not a valid URL github url. It should start with https://github.com/<user>/<repo>",
+                    githubUrl));
+        }
+
+        // ok so now get matching group
+        int regexpGroupCount = matcher.groupCount();
+        if (regexpGroupCount < 2) {
+            throw new BadRequestException(String.format(
+                    "The given github url %s is not a valid URL github url. It should start with https://github.com/<user>/<repo>",
+                    githubUrl));
+        }
+
+        String repoUser = matcher.group(1);
+        String repoName = matcher.group(2);
+        String branchName = null;
+        String subfolder = null;
+        if (regexpGroupCount == 4) {
+            branchName = matcher.group(3);
+            subfolder = matcher.group(4);
+        }
+
+        // build factory for the given github url
+        Map<String, String> parameters = new HashMap<>();
+        if (branchName != null) {
+            parameters.put("branch", branchName);
+        }
+        if (subfolder != null && !"/".equals(subfolder)) {
+            parameters.put("keepDir", subfolder);
+        }
+
+        // github location for the clone
+        String githubLocation = "https://github.com/".concat(repoUser).concat("/").concat(repoName);
+
+
+        // default
+        final String DEFAULT_DOCKER_FILE_LOCATION = "https://dockerfiles.codenvycorp.com/templates-4.0/factory/factory-dockerfile";
+
+        // use master if not specified
+        if (branchName == null) {
+            branchName = "master";
+        }
+
+        // check if repository contains a codenvy Dockerfile
+        String dockerFileLocation = null;
+        String githubDockerFileLocation =
+                "https://raw.githubusercontent.com/".concat(repoUser).concat("/").concat(repoName).concat("/").concat(branchName)
```

Comment:
Consider usage of plus operator. It is more expressive, shorter and can be optimized